### DISCUSSION TITLE
Is Doom full version supported?

### DISCUSSION THREAD

**Discussion by pazuzu72** [frntc/RAD-Doom#2] [General]
I don't know if the info is available somewhere but I can't find it. I can only run the shareware version of DOOM, not the full version. It's right?

Thanks.

**Comment by frntc** [comment]
I could only test it with the shareware version as I don't have a full version. I think something (like paths, filenames) would need to be changed in the 'generic doom'-files included in RAD-Doom.

**Reply by pazuzu72** [reply]
Ok, so no, without recompiling it.
Thank you

**Comment by hpingel** [comment]
I did some testing with a slightly modified RAD-Doom kernel.
![IMG_20231217_022855](https://github.com/frntc/RAD-Doom/assets/646865/5cdcd1f8-48eb-4139-b486-acc970aea74b)
![IMG_20231214_222519~2](https://github.com/frntc/RAD-Doom/assets/646865/375336b4-8317-4758-a561-f7ec331e25c7)
![IMG_20231214_225958](https://github.com/frntc/RAD-Doom/assets/646865/69960309-fb5d-45da-b373-b03a03770445)

**Reply by pazuzu72** [reply]
Coool!!!!!

**Reply by danielditlev** [reply]
Hmm .. I have the selection menu sorted but struggle with getting the DoomII wad to run. It bleebs and then freezes. Any hints to how you made it run?

> In regards to the classic Doom WADs - Doom II (currently on offer at gog.com - I bought this a couple of days ago to have a licensed copy of Doom II https://www.gog.com/de/game/doom_ii ) looks quite good but if you look at my Plutonia photo above it doesn't look appealing. It is possible that some WADs use color schemes in their levels that look less good on C64.

**Reply by danielditlev** [reply]
Okay, i figured it out. I have both a menu selection and DoomII running ( and joystick support for that matter). I will share the code during Easter, will just need to clean up all the debug code.

**Comment by pazuzu72** [comment]
Are you planning on releasing it?

**Reply by hpingel** [reply]
As I have no confirmation that anybody out there has actually played the shareware version doom1.wad for more than 2 minutes on  C64 with RAD (except for myself - LOL) I doubt that a release would create value for anybody. :-)

I hacked the sourcecode because I wanted to check if I could get the brand new Romero levels of Sigil II to work. Doing this I learned a lot about WADs (IWAD vs. PWAD) and learned a lot about different Doom source ports. Sadly it will take time until all the different Doom source ports will be able to run Sigil II. ATM only gzdoom can run it and many other Doom engines have to be adopted for Sigil II. So by researching the current activities of the Doom scene I created a half-broken IWAD that contains Sigil II. The levels run on RAD but there are issues with the music assigned (wrong music for the new levels). So under no circumstances there is a quick solution for Sigil II. We have to wait for the Doom experts to adopt their engines for Sigil II. This  will take time. And it has to be ported to doomgeneric. (Also doomgeneric can't merge an IWAD and a PWAD as other engines can.)

In regards to the classic Doom WADs - Doom II (currently on offer at gog.com  - I bought this a couple of days ago to have a licensed copy of Doom II https://www.gog.com/de/game/doom_ii ) looks quite good but if you look at my Plutonia photo above it doesn't look appealing. It is possible that some WADs use color schemes in their levels that look less good on C64.

If I release something and it doesn't look good and there is no visual menu to select a desired WAD in a user-friendly way then there is no value added. :-)

**Comment by danielditlev** [comment]
Here you go. Just add the Doom, Doom SW, DoomII, or any of the other WADs that generic doom supports to the RADDOOM folder and use this kernel in your config file. This will allow you to run the mentioned wads and add a nice simple menu to allow you to select between them. It also adds joystick support in port 2 (you can still switch to mouse mode if you want to). I haven't tested all the possible wads but all of what I have tested has worked. 

updated to latest version:
[kernel_dd.img.zip](https://github.com/frntc/RAD-Doom/files/14817826/kernel_dd.img.zip)

You can see an early version of this in action here : https://youtu.be/KYWlnlr1Faw

**Reply by frntc** [reply]
Nice!

**Reply by pazuzu72** [reply]
Great. Tried Doom 2 quickly seems to go. Thanks so much for your contribution.

**Reply by danielditlev** [reply]
Thank you. It is an amazing project and very cool concept. Hats off to @frntc !

**Comment by danielditlev** [comment]
Updated the code to enable an easier to read UI / HUD when playing (press 'u' to toogle it)  + an updated menu.

Can be seen in action here (updated video) : https://youtu.be/mr99G6hdYrM?si=AA6FXu_O5pWHmJow

Updated kernel is here (updated again) : 
[kernel_dd.img.zip](https://github.com/frntc/RAD-Doom/files/14961085/kernel_dd.img.zip)

**Comment by HardwareHuligan** [comment]
do we have  working Save  option in doom1&2  ?
